universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java

```diff
@@ -0,0 +1,100 @@
+package views.admin.questions;
+
+import static j2html.TagCreator.body;
+import static j2html.TagCreator.br;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.label;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import j2html.tags.ContainerTag;
+import j2html.tags.DomContent;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Optional;
+import play.mvc.Http.Request;
+import play.twirl.api.Content;
+import services.question.QuestionDefinition;
+import services.question.QuestionType;
+import services.question.TranslationNotFoundException;
+import views.BaseHtmlLayout;
+import views.BaseHtmlView;
+
+public final class QuestionEditView extends BaseHtmlView {
+  private final BaseHtmlLayout layout;
+
+  @Inject
+  public QuestionEditView(BaseHtmlLayout layout) {
+    this.layout = layout;
+  }
+
+  public Content render(Request request, Optional<QuestionDefinition> question) {
+    String headerText = question.isPresent() ? "Edit Question" : "New Question";
+    return layout.htmlContent(
+        body(
```

Comment:
question: should this be `body().with()` to be consistent with elsewhere? I am just curious if they are the same or there is some difference there.

Reply:
They're _mostly_ the same. 

If the tag that you're using is a `ContainerTag` (which most are) then you can use `body().with(tag).with(tag2)` or `body(tag1, tag2)`

The base tag class doesn't implement `with` but we'll likely never be using that unless we run into an unsupported type.

I try to use `.with` because I think it's cleaner, but that's not always true (and can largely depend on what the auto-formatter does to your code.)